eng/common/scripts/Invoke-GitHub-API.ps1

```diff
@@ -0,0 +1,180 @@
+      -Body ($body | ConvertTo-Json) `
+      -Uri $apiURI `
+      -Authentication Bearer `
+      -Token $Token `
+      -MaximumRetryCount 3
+
+  return $resp
+}
+
+function InvokeGet($apiURI) {
+  $resp = Invoke-RestMethod `
+      -Method GET `
+      -Uri $apiURI `
+      -Authentication Bearer `
+      -Token $Token `
+      -MaximumRetryCount 3
+      
+  return $resp
+}
+
+function SplitMembers ($membersString)
+{
+  if (!$membersString) { return $null }
+  return @($membersString.Split(",") | % { $_.Trim() } | ? { return $_ })
+}
+
+function ListPullRequests {
+  param (
+    [Parameter(Mandatory = $true)]
+    $RepoOwner,
+    [Parameter(Mandatory = $true)]
+    $RepoName,
+    [ValidateSet("open","closed","all")]
+    $state = "open",
+    $head,
+    $base,
+    [ValidateSet("created","updated","popularity","long-running")]
+    $sort,
+    [ValidateSet("asc","desc")]
+    $direction
+  )
+
+  $uri = "$GithubAPIBaseURI/$RepoOwner/$RepoName/pulls"
+  if ($state -or $head -or $base -or $sort -or $direction) { $uri += '?'}
+  if ($state){ $uri += "state=$state&" }
+  if ($head){ $uri += "head=$head&" }
+  if ($base){ $uri += "base=$base&" }
+  if ($sort){ $uri += "sort=$sort&" }
+  if ($direction){ $uri += "direction=$direction&" }
+
+  return InvokeGet -apiURI $uri
+}
+
+function AddIssueComment {
+  param (
+    [Parameter(Mandatory = $true)]
+    $RepoOwner,
+    [Parameter(Mandatory = $true)]
+    $RepoName,
+    [Parameter(Mandatory = $true)]
+    $IssueNumber,
+    $CommentPrefix,
+    [Parameter(Mandatory = $true)]
+    $Comment,
+    $CommentSuffix
+  )
+  $uri = "$GithubAPIBaseURI/$RepoOwner/$RepoName/issues/$IssueNumber/comments"
+  $PRComment = "$CommentPrefix $Comment $CommentSuffix"
+
+  $parameters = @{
+    body = $PRComment
+  }
+  return InvokePost -apiURI $uri -body $parameters
+}
+
+# Will add labels to existing labels on the issue
+function AddIssueLabels {
+  param (
+    [Parameter(Mandatory = $true)]
+    $RepoOwner,
+    [Parameter(Mandatory = $true)]
+    $RepoName,
+    [Parameter(Mandatory = $true)]
+    $IssueNumber,
+    [ValidateNotNullOrEmpty()]
+    [Parameter(Mandatory = $true)]
+    $labels
+  )
+
+  $uri = "$GithubAPIBaseURI/$RepoOwner/$RepoName/issues/$IssueNumber/labels"
+  $labelAdditions = SplitMembers -membersString $labels
+  $parameters = @{
+    labels = @($labelAdditions)
+  }
+
+  return InvokePost -apiURI $uri -body $parameters
+}
+
+# Will add assignees to existing assignees on the issue
+function AddIssueAssignees {
+  param (
+    [Parameter(Mandatory = $true)]
+    $RepoOwner,
+    [Parameter(Mandatory = $true)]
+    $RepoName,
+    [Parameter(Mandatory = $true)]
+    $IssueNumber,
+    [ValidateNotNullOrEmpty()]
+    [Parameter(Mandatory = $true)]
+    $assignees
+  )
+
+  $uri = "$GithubAPIBaseURI/$RepoOwner/$RepoName/issues/$IssueNumber/assignees"
+  $assigneesAdditions = SplitMembers -membersString $assignees
+  $parameters = @{
+    assignees = @($assigneesAdditions)
```

Comment:
Doesn't split members already return a array? Do we need to wrap it in another array?

Reply:
Should we also verify that the assignees is not null/empty? Or do we want to support case as I suspect it will remove the assignees.

Reply:
I have `[ValidateNotNullOrEmpty()]` to take care of that.

Reply:
Isn't setting it to empty a potentially valid thing to do if you want to clear the assignee?

Reply:
Not for this method. To o that one can use the `Update-Issue` method. Which is an `PATCH` rather than `POST` method.